src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -84,7 +84,14 @@ private[snowflake] class JDBCWrapper {
         val isSigned = rsmd.isSigned(i + 1)
         val nullable = rsmd.isNullable(i + 1) != ResultSetMetaData.columnNoNulls
         val columnType = getCatalystType(dataType, fieldSize, fieldScale, isSigned)
-        fields(i) = StructField(columnName, columnType, nullable)
+        fields(i) =
+          StructField(
+            if(columnName == columnName.toUpperCase) // Add quotes around column names if not all uppercase
```

Comment:
Good point. I now pattern match for all identifiers (after upcasting) allowed by Snowflake without quotes (as defined here: https://docs.snowflake.net/manuals/sql-reference/identifiers.html), and then add quotes to names that don't match during DataFrame column resolution. The user should now have to manually quote (in Spark) all identifiers where they would normally have to when directly using SQL against Snowflake.